packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2915,6 +2916,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             throw new UsageError("can't get state when offline load disabled");
         }
 
+        const pendingMessages = this.pendingStateManager.getLocalState();
+        const firstMessage = pendingMessages?.pendingStates[0] as IPendingMessage;
+        const isFirstMessageBatchFirst = firstMessage.opMetadata?.batch;
+        const lastMessage = pendingMessages?.pendingStates[pendingMessages.pendingStates.length - 1].type;
+        assert(!!isFirstMessageBatchFirst, "First message must have batch metadata");
```

Comment:
These look like some invariants which are fine to check for. But from the perspective of a client of this particular (public) method, I don't know what's that all about, as the errors are leaking some information about batch mechanics which do not seem relevant. 

I think this would be better as a usage error (which we should also log) which can record the values of these conditions in the telemetry properties of the event when it's logged. Like if any of these are not met, throw a usage error like 'Cannot call this while processing a batch of ops' (and log) the error which would have `isFirstMessageBatchFirst, lastStateFlush: lastMessage === "flush"` as telemetry properties.